### ISSUE TITLE
Assertion errors while receiving pings

### ISSUE THREAD

**Issue by holiman** [ethereum/pydevp2p#34]
I installed pyethapp+pyethereum in a brand new virtualenv, installed from git. Started synching the blockchain, got the following errors. 

It didn't stop the client, it keeps chugging along, but it indicates some error somewhere, so here you go: 

```
Traceback (most recent call last):
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/greenlet.py", line 519, in run
    result = self._run(*self.args, **self.kwargs)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/baseserver.py", line 153, in _close_when_done
    return handle(*args)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 560, in _handle_packet
    self.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 554, in receive
    self.protocol.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 322, in receive
    cmd(nodeid, payload, mdc)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 383, in recv_ping
    remote_address = Address.from_endpoint(*payload[1])  # from address
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 103, in from_binary
    return Address(ip, udp_port, tcp_port, from_binary=True)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 49, in __init__
    assert len(ip) in (4, 16), repr(ip)
AssertionError: '188.134.17.251'
<Greenlet at 0x7f7dd25707d0: _close_when_done('\x1d\xf3j\x8e8\xd9\x82\x19&\xca>x(\x9e\xe6\xfc\xc, ('188.134.17.251', 30303))> failed with AssertionError
```

and

```
WARNING:eth.chainservice    invalid network id remote_network_id=996787477 expected_network_id=1
WARNING:protocol    protocol exception, stopping error=wrong network_id
Traceback (most recent call last):
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/greenlet.py", line 519, in run
    result = self._run(*self.args, **self.kwargs)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/baseserver.py", line 153, in _close_when_done
    return handle(*args)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 560, in _handle_packet
    self.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 554, in receive
    self.protocol.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 322, in receive
    cmd(nodeid, payload, mdc)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 413, in recv_pong
    assert len(payload[0][0]) in (4, 16), payload
AssertionError: [['h-37-153.a328.priv.bahnhof.se', 'v_', 'v_'], '\x13\xd1O\xd0s\xca\x9c\xeb\xab\x1a(\xc1\x05E\xb20[\xc1\x92\x1c\xfan\xafr\xe4\xad)\xc8\x9f\xeaP\xbb']
<Greenlet at 0x7f7dd31a79b0: _close_when_done('\xa4\xe8\xc9\xceg\x99\xba#\x95\xed\xacJ2\x8a\x05\, ('104.236.197.73', 30303))> failed with AssertionError
```

**Comment by holiman** [comment]
Also when receiving pong:

```
Traceback (most recent call last):
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/greenlet.py", line 519, in run
    result = self._run(*self.args, **self.kwargs)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/gevent/baseserver.py", line 153, in _close_when_done
    return handle(*args)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 560, in _handle_packet
    self.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 554, in receive
    self.protocol.receive(address, message)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 322, in receive
    cmd(nodeid, payload, mdc)
  File "/data/workspace/pyethapp/.venv/local/lib/python2.7/site-packages/devp2p/discovery.py", line 413, in recv_pong
    assert len(payload[0][0]) in (4, 16), payload
AssertionError: [['46.59.37.153', 'v_', 'v_'], '\x96U\xe5\xd6\xe1\x03\x8f*}\xad\xe66Q<\xb9\xec\x05\x9d\x94k\xf1\tQ\xfc\xc7\xeb\n\xb1tk\x9a\n']
<Greenlet at 0x7f7dd3fad050: _close_when_done('t`\x16\xc9\xa0\xbb\xdc\xcf\x8aw\x94\x80\x16\x18AS, ('78.46.49.102', 30303))> failed with AssertionError
```

**Comment by heikoheiko** [comment]
The IP 46.59.37.153 and hostname h-37-153.a328.priv.bahnhof.se are these related to your host?

I still wonder which client is sending this format.

**Comment by holiman** [comment]
That's correct, those are the public ip-addresses from my ISP. Can't imagine they come from me, since my computer don't know about them (NAT).

**Comment by holiman** [comment]
Do you mean which client as in "which software-implementation" or as in "me or another peer" ? I'm guessing you mean the former.

**Comment by 4gn3s** [comment]
Did you manage to fix the problem? I'm getting the same errors (different IPs) in a brand new virtualenv.

**Comment by awdiama** [comment]
seeing same issue every time I run client too

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent/greenlet.py", line 534, in run
    result = self._run(_self.args, *_self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/gevent/baseserver.py", line 25, in _handle_and_close_when_done
    return handle(*args_tuple)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 565, in _handle_packet
    self.receive(address, message)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 559, in receive
    self.protocol.receive(address, message)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 331, in receive
    cmd(nodeid, payload, mdc)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 418, in recv_pong
    assert len(payload[0][0]) in (4, 16), payload
AssertionError: [['87.115.202.139', 'v_', 'v_'], '\x02\xff\x8f\xf6\x03\xad[\xb4\xac\x96n1\x88\xb2\x12\xf0`\xb2\xd5U`-\xed\r\x05\xb12\xa3\x00\x0b-D']
<Greenlet at 0x7f270cdce910: _handle_and_close_when_done(<bound method NodeDiscovery._handle_packet of <Nod, <bound method DatagramServer.do_close of <Datagram, ("l\x93\xf8t1\xd0\xe9\xc2f@\xf7c\x178\xdd=\xf0\x9e)> failed with AssertionError

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent/greenlet.py", line 534, in run
    result = self._run(_self.args, *_self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/gevent/baseserver.py", line 25, in _handle_and_close_when_done
    return handle(_args_tuple)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 565, in _handle_packet
    self.receive(address, message)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 559, in receive
    self.protocol.receive(address, message)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 331, in receive
    cmd(nodeid, payload, mdc)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 388, in recv_ping
    remote_address = Address.from_endpoint(_payload[1])  # from address
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 103, in from_binary
    return Address(ip, udp_port, tcp_port, from_binary=True)
  File "/usr/local/lib/python2.7/dist-packages/devp2p/discovery.py", line 49, in __init__
    assert len(ip) in (4, 16), repr(ip)
AssertionError: '13.79.154.168'
<Greenlet at 0x7f270cdce870: _handle_and_close_when_done(<bound method NodeDiscovery._handle_packet of <Nod, <bound method DatagramServer.do_close of <Datagram, ('\x06U\xa4(z;\xaf\xad\xe2\xc5.\xd3\x99\xbb\xadg\x)> failed with AssertionError

**Comment by ulope** [comment]
The fix has been incorporated into PyEthApp 1.2.1
